core/domain/exp_services_test.py

```diff
@@ -881,13 +881,132 @@ def test_export_by_versions(self):
 def _get_change_list(state_name, property_name, new_value):
     """Generates a change list for a single state change."""
     return [{
-        'cmd': 'edit_state_property',
+        'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
         'state_name': state_name,
         'property_name': property_name,
         'new_value': new_value
     }]
 
 
+class UpdateGadgetTests(ExplorationServicesUnitTests):
+    """Test updating a gadget."""
+
+    def setUp(self):
+        super(UpdateGadgetTests, self).setUp()
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id)
+        # Default outcome specification for an interaction.
+        self.init_state_name = exploration.init_state_name
+
+        self.interaction_default_outcome = {
+            'dest': self.init_state_name,
+            'feedback': ['Incorrect', '<b>Wrong answer</b>'],
+            'param_changes': []
+        }
+
+    def test_add_gadget_cmd(self):
+        """test adding of a gadget"""
+        gadget_customization = {
+            'adviceObjects': {
+                'value': [{
+                    'adviceTitle': 'b',
+                    'adviceHtml': '<p>c</p>'
+                }]
+            }
+        }
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+
+        self.assertNotIn('NewGadget', exploration.get_all_gadget_names())
+        exp_services.update_exploration(self.owner_id, self.EXP_ID, [{
+            'cmd': exp_domain.CMD_ADD_GADGET,
+            'panel': 'bottom',
+            'gadget_dict' : {
+                'gadget_type': 'ScoreBar',
+                'gadget_name': 'NewGadget',
+                'customization_args': gadget_customization,
```

Comment:
You could just inline ``gadget_customization``, I'm not sure it really needs to be its own variable (we're not reusing anywhere else in the test, for instance).